wp-content/themes/pub/wporg-learn-2024/src/lesson-facilitator-notes/index.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * WordPress dependencies
+ */
+import { registerBlockType } from '@wordpress/blocks';
+import { RichText, useBlockProps } from '@wordpress/block-editor';
+import { Button, ComboboxControl, Icon, Spinner } from '@wordpress/components';
+import { check } from '@wordpress/icons';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+import './style.scss';
+
+registerBlockType( metadata.name, {
+	edit: function Edit( { attributes, setAttributes } ) {
+		const { lessonPlanId, lessonPlanContent, lessonPlanTitle } = attributes;
+		const [ searchResults, setSearchResults ] = useState( [] );
+		const [ isExpanded, setIsExpanded ] = useState( false );
+		const [ isSaving, setIsSaving ] = useState( false );
+		const [ saveSuccess, setSaveSuccess ] = useState( null );
+		const [ errorMessage, setErrorMessage ] = useState( null );
+
+		useEffect( () => {
+			// Fetch the initial lesson plan options if lessonPlanId is set
+			if ( lessonPlanId ) {
+				apiFetch( {
+					path: `/wp/v2/lesson-plan/${ lessonPlanId }`,
+				} ).then( ( plan ) => {
+					setSearchResults( [
+						{
+							value: plan.id,
+							label: plan.title.rendered,
+						},
+					] );
+				} );
+			}
+		}, [] );
+
+		const fetchLessonPlanContent = ( id ) => {
+			apiFetch( { path: `/wp/v2/lesson-plan/${ id }` } ).then( ( plan ) => {
+				const cleanedContent = plan.content.rendered.replace( /\s+/g, ' ' ).trim();
+				setAttributes( {
+					lessonPlanContent: cleanedContent,
+					lessonPlanTitle: plan.title.rendered,
+				} );
+			} );
+		};
+
+		const fetchLessonPlans = debounce( ( searchTerm ) => {
+			apiFetch( {
+				path: `/wp/v2/lesson-plan?search=${ encodeURIComponent( searchTerm ) }&per_page=10`,
```

Comment:
I had a lot of trouble finding specific lesson plans in my local site - I was typing exact keywords in the lesson title, but nothing would show up. I think because this search is default WP search, so it's returning when the keywords match in title or content, but the combobox control only matches results on title.

So the API is limited to 10 items, and my search "block directory" returned more than 10 items, pushing the page I wanted out of the options. And nothing would show up when I typed, because "block directory" is not a match in any of the results I had:

- Create a Basic Child Theme for Block Themes
- WebP images in WordPress
- Υποβολή Μοτίβων μπλοκ (Block Patterns) στον κατάλογο.
- Πώς να χρησιμοποιήσετε και να συνεισφέρετε στον κατάλογο φωτογραφιών του WordPress αλλά και στο Openverse
- Submitting Block Patterns to the Directory
- Template Editor
- How to Use and Contribute to the WordPress Photo Directory and Openverse
- How to Build Low-Code Block Patterns
- How to Create and Register a Block Pattern
- Difference between Reusable Blocks,  Block Pattern, Templates, Template Parts

Maybe increase the result number? Or see if you can adjust the search for this endpoint to only do title searches.

You could preload the ID/title pairs, and not use an API request for this at all. With 150 published lesson plans, it shouldn't be _that_ slow. Or if you'd rather not do that on the server side, fire off an API request to get all the lesson plans when the block loads (instead of just `if ( lessonPlanId )`).